managed/services/server/server.go

```diff
@@ -895,7 +897,15 @@ func isAgentsStateUpdateNeeded(mr *serverpb.MetricsResolutions) bool {
 	return true
 }
 
-// check interfaces
+// check interfaces.
 var (
 	_ serverpb.ServerServer = (*Server)(nil)
 )
+
+func canUpdateDurationSetting(valueToBeSet, valueFromEnvironmentVariable time.Duration) bool {
+	if valueToBeSet == 0 || valueFromEnvironmentVariable == 0 || valueToBeSet == valueFromEnvironmentVariable {
+		return true
+	}
+
+	return false
```

Comment:
it can even be simplified like
```suggestion
	return valueToBeSet == 0 || valueFromEnvironmentVariable == 0 || valueToBeSet == valueFromEnvironmentVariable
```